ksql-execution/src/main/java/io/confluent/ksql/execution/plan/StreamSelectKey.java

```diff
@@ -22,18 +24,28 @@
 
 @Immutable
 public class StreamSelectKey<S> implements ExecutionStep<S> {
+  private static final String PROPERTIES = "properties";
+  private static final String SOURCE = "source";
+  private static final String FIELD_NAME = "fieldName";
+  private static final String UPDATE_ROW_KEY = "updateRowKey";
+
+  @JsonProperty(PROPERTIES)
   private final ExecutionStepProperties properties;
+  @JsonProperty(SOURCE)
   private final ExecutionStep<S> source;
+  @JsonProperty(FIELD_NAME)
   private final String fieldName;
+  @JsonProperty(UPDATE_ROW_KEY)
   private final boolean updateRowKey;
 
+  @JsonCreator
   public StreamSelectKey(
-      final ExecutionStepProperties properties,
-      final ExecutionStep<S> source,
-      final String fieldName,
-      final boolean updateRowKey) {
+      @JsonProperty(PROPERTIES) final ExecutionStepProperties properties,
+      @JsonProperty(SOURCE) final ExecutionStep<S> source,
+      @JsonProperty(FIELD_NAME) final String fieldName,
+      @JsonProperty(UPDATE_ROW_KEY) final boolean updateRowKey) {
     this.properties = Objects.requireNonNull(properties, "properties");
-    this.source = Objects.requireNonNull(source, "source");
+    this.source = source;  // Objects.requireNonNull(source, "source");
```

Comment:
Why is Objects.requireNonNull() commented? If null is allowed, should we use Optional? should we remove the commented line?